src/change_stream.ts

```diff
@@ -648,21 +648,21 @@ export class ChangeStream<
   hasNext(callback?: Callback): Promise<boolean> | void {
     this._setIsIterator();
     return maybeCallback(async () => {
-      try {
-        const hasNext = await this.cursor.hasNext();
-        return hasNext;
-      } catch (error) {
+      for (;;) {
```

Comment:
This is less obvious than a while true loop, so unless there's a particularly good reason to use this, I'd prefer the while loop (yes even with the lint warning, maybe even with a comment describing why this is definitely intentional)